docs/openstack/nova_adoption.md

```diff
@@ -0,0 +1,176 @@
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novaapis --field-selector metadata.name=nova-api -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novacells --field-selector metadata.name=nova-cell0 -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novacells --field-selector metadata.name=nova-cell1 -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novaconductors --field-selector metadata.name=nova-cell0-conductor -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novaconductors --field-selector metadata.name=nova-cell1-conductor -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novametadata --field-selector metadata.name=nova-metadata -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novanovncproxies --field-selector metadata.name=nova-cell1-novncproxy -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  oc get novaschedulers --field-selector metadata.name=nova-scheduler -o jsonpath='{.items[0].status.conditions}' \
+  | jq -e '.[]|select(.type=="Ready" and .status=="True")'
+  ```
+
+  The local Conductor services will be started for each cell, while the superconductor runs in `cell0`.
+  Note that ``disable_compute_service_check_for_ffu`` is mandatory for all imported Nova services, until
+  the [external dataplane imported](edpm_adoption.md), and until Nova Compute services fast-forward upgraded.
+
+## Post-checks
+
+* Check that Nova endpoints are defined and pointing to the
+  podified FQDNs and that Nova API responds.
+
+  ```bash
+  openstack endpoint list | grep nova
+  openstack server list
+  ```
+
+Compare the following outputs with the topology specific configuration
+[collected earlier](pull_openstack_configuration.md#get-services-topology-specific-configuration):
+
+* Query the superconductor for cell1 existance:
+
+  ```bash
+  oc rsh nova-cell0-conductor-0 nova-manage cell_v2 list_cells | grep -F '| cell1 |'
+  ```
+
+  The expected changes to happen:
+  * cell1's `nova` DB and user name become `nova_cell1`.
+  * Default cell is renamed to `cell1` (in a multi-cell setup, it should become indexed as the last cell instead).
+  * RabbitMQ transport URL no longer uses `guest`.
+
+* Verify no Nova compute dataplane disruptions during the adoption/upgrade process:
+
+  ```bash
+  $CONTROLLER_SSH sudo podman exec -it libvirt_virtqemud virsh list --all | grep 'instance-00000001   running'
+  ```
+
+  * Verify if Nova services control the existing VM instance:
+
+  ```bash
+  openstack server list | grep -qF '| test | ACTIVE |' && openstack server stop test
```

Comment:
These steps should not work at this point as the compute node is not yet adopted so there would be no way to stop a VM as the compute is still listening on the old message bus.

Reply:
At this point we expect that the podified control plane see the adopted DB service record for the compute but that compute is in down state as it never reported status to the new control plane:
```
[gibi@osp-dev-01 ~]$ openstack compute service list
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| ID                                   | Binary         | Host                   | Zone     | Status  | State | Updated At                 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| 0954e21c-9022-4718-a570-a7d3eb0fd79f | nova-conductor | nova-cell0-conductor-0 | internal | enabled | up    | 2023-11-09T09:14:33.000000 |
| c3ad2def-18ed-49e5-8af4-a7c1a0840171 | nova-scheduler | nova-scheduler-0       | internal | enabled | up    | 2023-11-09T09:14:29.000000 |
| a7a20d50-b85d-4321-a576-5a12fea9bc8f | nova-compute   | standalone.localdomain | nova     | enabled | down  | 2023-11-09T08:18:09.000000 |
| 9eb053f9-a404-4b02-92a8-d2a5fe339849 | nova-conductor | nova-cell1-conductor-0 | internal | enabled | up    | 2023-11-09T09:14:25.000000 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
[gibi@osp-dev-01 ~]$ openstack hypervisor list
+--------------------------------------+------------------------+-----------------+-----------------+-------+
| ID                                   | Hypervisor Hostname    | Hypervisor Type | Host IP         | State |
+--------------------------------------+------------------------+-----------------+-----------------+-------+
| d3d2be51-a0b9-4538-a298-62280a52fece | standalone.localdomain | QEMU            | 192.168.122.100 | down  |
+--------------------------------------+------------------------+-----------------+-----------------+-------+
```

Reply:
Done https://github.com/openstack-k8s-operators/data-plane-adoption/compare/9844aa6e67b34e3dbffe3734897722619619b260..762b0ea97b0915fb3dbef78ff96e8570e5636b33